### PR TITLE
improved debugging stats

### PR DESCRIPTION
**Pull request by ZappoMan**
Make sure the stats include the type and name of entity. And sort them for easier readability.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: e0b36b63e5fe8d07b9d1f4f720af1eb14b038521 available at: https://gist.github.com/059e611ca737dffa8b5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.dmg

File: assignment-client/src/entities/EntityServer.cpp [hunk 1]
```diff
@@ -272,9 +272,9 @@ void EntityServer::readAdditionalConfiguration(const QJsonObject& settingsSectio
 // set of stats to have, but we'd probably want a different data structure if we keep it very long.
 // Since this version uses a single shared QMap for all senders, there could be some lock contention 
 // on this QWriteLocker
-void EntityServer::trackSend(const QUuid& dataID, quint64 dataLastEdited, const QUuid& viewerNode) {
+void EntityServer::trackSend(const QUuid& dataID, quint64 dataLastEdited, QString dataType, const QUuid& viewerNode) {
```
Comment (jherico) [inline assignment-client/src/entities/EntityServer.cpp]: `const QString&`

**Comment by hifi-gustavo** [comment]
Build console output for commit: fa9eb99de47edfbac52cf2fb4ca35ec76ed01113 available at: https://gist.github.com/059e611ca737dffa8b5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.dmg

File: assignment-client/src/entities/EntityServer.cpp (hunk 1 above)
Comment (ZappoMan) [inline assignment-client/src/entities/EntityServer.cpp]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3054673de7d56b45f348642fd369ecbef8274286 available at: https://gist.github.com/059e611ca737dffa8b5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.dmg

File: libraries/octree/src/Octree.h [hunk 2]
```diff
@@ -177,7 +177,8 @@ class EncodeBitstreamParams {
         }
     }
 
-    std::function<void(const QUuid& dataID, quint64 itemLastEdited)> trackSend { [](const QUuid&, quint64){} };
+    std::function<void(const QUuid& dataID, quint64 itemLastEdited, QString dataType)> trackSend 
+                { [](const QUuid&, quint64, QString dataType){} };
```
Comment (Atlante45) [inline libraries/octree/src/Octree.h]: Aliasing the type and defining the default would make this a bit more readable I think.
Something like:
```cpp
using TrackingCallback = std::function<void(const QUuid& dataID, quint64 itemLastEdited, QString dataType)>;
static const TrackingCallback DEFAULT_TRACKING_CALLBACK = [](const QUuid&, quint64, QString) {};

TrackingCallback trackSend { DEFAULT_TRACKING_CALLBACK };
```

Comment (ZappoMan) [inline libraries/octree/src/Octree.h]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 92535dbcd6e188d3c9e9b7a03795e4015a3a2555 available at: https://gist.github.com/059e611ca737dffa8b5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa41030f5d95fa0358d21d28f33c1dcc3d31cb85 available at: https://gist.github.com/448d41297067c6f620c5<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6533.dmg

**Pull request closed without merging**